parsec/backend/memory/vlob.py

```diff
@@ -101,33 +165,67 @@ def __init__(self, send_event):
         self._vlobs = {}
         self._per_realm_changes = defaultdict(Changes)
 
-    def register_components(self, realm: BaseRealmComponent, **other_components):
+    def register_components(
+        self, realm: BaseRealmComponent, blockstore: BaseBlockStoreComponent, **other_components
+    ):
         self._realm_component = realm
+        self._blockstore_component = blockstore
 
-    def _maintenance_reencryption_start_hook(self, organization_id, realm_id, encryption_revision):
+    def _get_changes_to_maintenance_start_hook(self, organization_id, realm_id):
         changes = self._per_realm_changes[(organization_id, realm_id)]
         assert not changes.reencryption
-        realm_vlobs = {
+        assert not changes.garbage_collection
+        return changes
+
+    def _get_realm_vlobs(self, organization_id, realm_id):
+        return {
             vlob_id: vlob
             for (orgid, vlob_id), vlob in self._vlobs.items()
             if orgid == organization_id and vlob.realm_id == realm_id
         }
-        changes.reencryption = Reencryption(realm_id, realm_vlobs)
 
-    def _maintenance_reencryption_is_finished_hook(
-        self, organization_id, realm_id, encryption_revision
-    ):
+    def _maintenance_start_hook(self, attr, Task, organization_id, realm_id, task_args=None):
+        task_args = task_args or []
+        changes = self._get_changes_to_maintenance_start_hook(organization_id, realm_id)
+        realm_vlobs = self._get_realm_vlobs(organization_id, realm_id)
+        setattr(changes, attr, Task(realm_id, realm_vlobs, *task_args))
```

Comment:
Tu penses que ça vaudrait le coup d'être paranoavec un `assert attr in ('reencryption', 'garbage_collection')` ?